zcash_keys/src/keys.rs

```diff
@@ -831,85 +889,129 @@ mod tests {
+            assert_eq!(encoded, reencoded);
 
-        #[cfg(feature = "transparent-inputs")]
-        assert_eq!(
-            decoded.transparent.map(|t| t.serialize()),
-            ufvk.transparent.as_ref().map(|t| t.serialize()),
-        );
-        assert_eq!(
-            decoded.sapling.map(|s| s.to_bytes()),
-            ufvk.sapling.map(|s| s.to_bytes()),
-        );
-        #[cfg(feature = "orchard")]
-        assert_eq!(
-            decoded.orchard.map(|o| o.to_bytes()),
-            ufvk.orchard.map(|o| o.to_bytes()),
-        );
+            #[cfg(feature = "transparent-inputs")]
+            assert_eq!(
+                decoded.transparent.map(|t| t.serialize()),
+                ufvk.transparent.as_ref().map(|t| t.serialize()),
+            );
+            #[cfg(feature = "sapling")]
+            assert_eq!(
+                decoded.sapling.map(|s| s.to_bytes()),
+                ufvk.sapling.map(|s| s.to_bytes()),
+            );
+            #[cfg(feature = "orchard")]
+            assert_eq!(
+                decoded.orchard.map(|o| o.to_bytes()),
+                ufvk.orchard.map(|o| o.to_bytes()),
+            );
 
-        let decoded_with_t = UnifiedFullViewingKey::decode(&MAIN_NETWORK, encoded_with_t).unwrap();
-        #[cfg(feature = "transparent-inputs")]
-        assert_eq!(
-            decoded_with_t.transparent.map(|t| t.serialize()),
-            ufvk.transparent.as_ref().map(|t| t.serialize()),
-        );
+            let decoded_with_t =
+                UnifiedFullViewingKey::decode(&MAIN_NETWORK, encoded_with_t).unwrap();
+            #[cfg(feature = "transparent-inputs")]
+            assert_eq!(
+                decoded_with_t.transparent.map(|t| t.serialize()),
+                ufvk.transparent.as_ref().map(|t| t.serialize()),
+            );
 
-        #[cfg(all(not(feature = "orchard"), not(feature = "transparent-inputs")))]
-        assert_eq!(decoded_with_t.unknown.len(), 2);
-        #[cfg(all(feature = "transparent-inputs", not(feature = "orchard")))]
-        assert_eq!(decoded_with_t.unknown.len(), 1);
-        #[cfg(all(feature = "orchard", not(feature = "transparent-inputs")))]
-        assert_eq!(decoded_with_t.unknown.len(), 1);
+            // Both Orchard and Sapling enabled
+            #[cfg(all(
+                feature = "orchard",
+                feature = "sapling",
+                feature = "transparent-inputs"
+            ))]
+            assert_eq!(decoded_with_t.unknown.len(), 0);
+            #[cfg(all(
+                feature = "orchard",
+                feature = "sapling",
+                not(feature = "transparent-inputs")
+            ))]
+            assert_eq!(decoded_with_t.unknown.len(), 1);
+
+            // Orchard enabled
+            #[cfg(all(
+                feature = "orchard",
+                not(feature = "sapling"),
+                feature = "transparent-inputs"
+            ))]
+            assert_eq!(decoded_with_t.unknown.len(), 2);
```

Comment:
If I'm counting the `not`s in the `cfg` correctly, this should be:
```suggestion
            assert_eq!(decoded_with_t.unknown.len(), 1);
```